lib/util/Components.js

```diff
@@ -63,14 +63,50 @@ Components.prototype.get = function(node) {
  * @param {Object} props Additional properties to add to the component.
  */
 Components.prototype.set = function(node, props) {
+  const popsAreEquivalent = (propA, propB) => {
+    if (propA.name === propB.name) {
+      if (!propA.allNames && !propB.allNames) {
+        return true;
+      } else if (propA.allNames.join('') === propB.allNames.join('')) {
+        return true;
+      }
+      return false;
+    }
+    return false;
+  };
+
+  const merge = (propsList, newPropList) => {
+    const propsToAdd = [];
+    newPropList.forEach(newProp => {
+      let newPropisAlreadyInTheList = false;
+      propsList.some(prop => {
+        if (popsAreEquivalent(prop, newProp)) {
+          newPropisAlreadyInTheList = true;
+        }
+        return newPropisAlreadyInTheList;
+      });
+      if (!newPropisAlreadyInTheList) {
+        propsToAdd.push(newProp);
+      }
+    });
+    return propsList.concat(propsToAdd);
+  };
+
   while (node && !this._list[this._getId(node)]) {
     node = node.parent;
   }
   if (!node) {
     return;
   }
   const id = this._getId(node);
+  let copyUsedPropTypes;
+  if (this._list[id]) {
+    copyUsedPropTypes = this._list[id].usedPropTypes && this._list[id].usedPropTypes.slice();
```

Comment:
Is it possible that `copyUsedPropTypes` is going to be assigned a value but that it will not be used?

The condition for assigning the value is `if (this._list[id])`
The condition for using the value is: `if (this._list[id] && props.usedPropTypes)`